### PR TITLE
Add tasks sample with Electron, React, TS

### PR DESCRIPTION
**Pull request by jreyes33**
Related issue: #6 

I bootstrapped this sample with Create React App and [electron-quick-start-typescript](https://github.com/electron/electron-quick-start-typescript), but removed as much code as possible to keep it simple.

The instructions in the README should work:
1. First run `npm install`
2. Add your app ID to `src/AppContainer.tsx`
3. Run `npm start`

image

I'll continue working on the same app but with JS instead of TS.

### CODE REVIEW

**Review by okdistribute — APPROVED** [review]
Looks great!

**Review by okdistribute — COMMENTED** [review]
Oops, sorry approved this pre-maturely and had a second-thought. This current implementation is using the javascript code and ditto is being initialized in the frontend. 

I think it would be more useful for production Electron applications to show an example which uses the Node.js backend, which would initialize ditto in the Electron main process. This would allow access to the Bluetooth and Wifi/LAN APIs and persistence, as well as be able to use the Electron app as a Ditto TCP server.

**Comment by mbalex99** [comment]
The renderer process should still be able to use BLE and LAN. @joaquindk confirm?

**Comment by jreyes33** [comment]
@mbalex99, yeah I need to disable the `contextIsolation` electron preference for that. I'm testing that right now but I'm running into this error:
```
'could not initialize thread_rng: Node.js crypto module is unavailable'
```

**Comment by jreyes33** [comment]
I've updated this PR to enable Node.js in the renderer process. You'll notice it working because offline storage in the filesystem is now enabled (a `ditto-tasks` directory is created), but I've yet to test other transports. It was necessary to stop using Create React App and set up Webpack ourselves instead; I tried to keep that config as simple as possible.

**Review by okdistribute — APPROVED** [review]

**Comment by joaquindk** [comment]
> The renderer process should still be able to use BLE and LAN. @joaquindk confirm?

@mbalex99 yes you should be able to access all of the core transports from the renderer process. This is the same thing we do on the DB Electron project, so perhaps @jreyes33 can look at that if the transports aren't working.

File: tasks/electron-react-ts/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  const [text, setText] = useState('')
+  const { ditto, documents: tasks } = usePendingCursorOperation<Task>({
+    collection: COLLECTION,
+  })
+  const { insert, removeByID, updateByID } = useMutations<Task>({
+    collection: COLLECTION,
+  })
+
+  useEffect(() => {
+    ditto?.startSync()
```
Comment (joaquindk) [inline tasks/electron-react-ts/src/App.tsx:20]: You'll probably want to stop syncing here once the component is unmounted.

Comment (jreyes33) [reply tasks/electron-react-ts/src/App.tsx:20]: That's right, thanks for pointing that out!

**Comment by joaquindk** [comment]
Ok glad to hear. CRA has historically always been more of a problem than a solution for us.

**Review by okdistribute — APPROVED** [review]

**Review by joaquindk — APPROVED** [review]

**Pull request merged**